src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -1150,6 +1152,58 @@ void Instance::HandleQueryIdentity(HandlerContext & ctx, const Commands::QueryId
 }
 #endif // CHIP_DEVICE_CONFIG_ENABLE_WIFI_PDC
 
+Instance * Instance::sInstanceRecord = nullptr;
```

Comment:
We should just use an `IntrusiveList` here instead of hand-rolling linked-list code, and also remove the cluster instance in `Shutdown`.

Reply:
I have tried IntrusiveList, but i'm confused how to resolve `curr != currentNetwork` in `DisconnectOtherNetworks` func with IntrusiveList. The `IteratorBase` from `IntrusiveList` can not compared with the `Instance*`. Any suggestions?

Reply:
I've tweaked the code to use InstrusiveList in this sub-PR: https://github.com/DejinChen/connectedhomeip/pull/1